src/Compilers/CSharp/Portable/CSharpResources.resx

```diff
@@ -5089,36 +5101,126 @@ To remove the warning, you can use /reference instead (set the Embed Interop Typ
   </data>
   <data name="ERR_EscapeVariable" xml:space="preserve">
     <value>Cannot use variable '{0}' in this context because it may expose referenced variables outside of their declaration scope</value>
   </data>
+  <data name="WRN_EscapeVariable" xml:space="preserve">
+    <value>Use of variable '{0}' in this context may expose referenced variables outside of their declaration scope</value>
+  </data>
+  <data name="WRN_EscapeVariable_Title" xml:space="preserve">
+    <value>Use of variable in this context may expose referenced variables outside of their declaration scope</value>
+  </data>
   <data name="ERR_EscapeCall" xml:space="preserve">
     <value>Cannot use a result of '{0}' in this context because it may expose variables referenced by parameter '{1}' outside of their declaration scope</value>
   </data>
   <data name="ERR_EscapeCall2" xml:space="preserve">
     <value>Cannot use a member of result of '{0}' in this context because it may expose variables referenced by parameter '{1}' outside of their declaration scope</value>
   </data>
+  <data name="WRN_EscapeCall" xml:space="preserve">
+    <value>Use of result of '{0}' in this context may expose variables referenced by parameter '{1}' outside of their declaration scope</value>
+  </data>
+  <data name="WRN_EscapeCall_Title" xml:space="preserve">
+    <value>Use of result in this context may expose variables referenced by parameter outside of their declaration scope</value>
+  </data>
+  <data name="WRN_EscapeCall2" xml:space="preserve">
+    <value>Use of member of result of '{0}' in this context may expose variables referenced by parameter '{1}' outside of their declaration scope</value>
+  </data>
+  <data name="WRN_EscapeCall2_Title" xml:space="preserve">
+    <value>Use of member of result in this context may expose variables referenced by parameter outside of their declaration scope</value>
+  </data>
   <data name="ERR_CallArgMixing" xml:space="preserve">
     <value>This combination of arguments to '{0}' is disallowed because it may expose variables referenced by parameter '{1}' outside of their declaration scope</value>
   </data>
+  <data name="WRN_CallArgMixing" xml:space="preserve">
+    <value>This combination of arguments to '{0}' may expose variables referenced by parameter '{1}' outside of their declaration scope</value>
+  </data>
+  <data name="WRN_CallArgMixing_Title" xml:space="preserve">
+    <value>This combination of arguments may expose variables referenced by parameter outside of their declaration scope</value>
+  </data>
   <data name="ERR_MismatchedRefEscapeInTernary" xml:space="preserve">
     <value>Branches of a ref conditional operator cannot refer to variables with incompatible declaration scopes</value>
   </data>
+  <data name="WRN_MismatchedRefEscapeInTernary" xml:space="preserve">
+    <value>The branches of the ref conditional operator refer to variables with incompatible declaration scopes</value>
```

Comment:
>The branches of the ref [](http://example.com/codeflow?start=11&length=23)

Consider matching the corresponding error "Branches of a ref conditional ..." since it doesn't seem necessary to differ.

Reply:
I would like the error and warning messages to match more closely, but in this case the sentence would not make sense (not proper English):
"Branches of a ref conditional operator refer to variables with incompatible declaration scopes"